crates/nu-command/src/system/run_external.rs

```diff
@@ -473,6 +473,15 @@ pub fn command_not_found(
         };
     }
 
+    // If we find a file, it's likely that the user forgot to set permissions
+    if Path::new(name).is_file() {
```

Comment:
Purely a question for @IanManske, as this is the same behavior used in the place where perform the `which` lookup (also just a `std::path::Path::new`) for externals:

Should we migrate to our internal types in `run-external` for pwd soundness?

Reply:
If so, I would be happy to do that in both places and tack it onto this PR :)